Runtime/Components/ConeRayAnglesEstimator/LocationBasedRayDataCollector.cs

```diff
@@ -0,0 +1,134 @@
+        /// The list of interactables whose gesture events this collector listens to.
+        /// </summary>
+        public List<HPUIBaseInteractable> Interactables { get => interactables; set => interactables = value; }
+
+        private XRHandJointID closestJoint;
+        private FingerSide closestSide;
+
+        /// <summary>
+        /// Starts data collection by subscribing to gesture events on all configured interactables.
+        /// </summary>
+        /// <returns>
+        /// The return value of the base <see cref="RaycastDataCollectorBase.StartDataCollection"/> call. A caller should
+        /// check this value to verify that collection was successfully started.
+        /// </returns>
+        public override bool StartDataCollection()
+        {
+            bool retval = base.StartDataCollection();
+
+            foreach (IHPUIInteractable interactable in Interactables)
+            {
+                interactable.GestureEvent.AddListener(OnGestureCallback);
+            }
+            return retval;
+        }
+
+        /// <summary>
+        /// Stops data collection by unsubscribing from gesture events on all configured interactables.
+        /// </summary>
+        /// <returns>
+        /// The return value of the base <see cref="RaycastDataCollectorBase.StopDataCollection"/> call.
+        /// </returns>
+        public override bool StopDataCollection()
+        {
+            foreach (IHPUIInteractable interactable in Interactables)
+            {
+                interactable.GestureEvent.RemoveListener(OnGestureCallback);
+            }
+
+            return base.StopDataCollection();
+        }
+
+        /// <summary>
+        /// Callback invoked when an interactable fires a gesture event.
+        ///
+        /// When a gesture transitions to the <see cref="HPUIGestureState.Stopped"/> state, the method groups the
+        /// samples currently saved in <c>currentInteractionData</c> by (hand joint, finger side), maps each group to the
+        /// appropriate <see cref="HPUIInteractorConeRayAngleSegment"/>, and appends a new
+        /// <see cref="ConeRayComputationDataRecord"/> for each group into <see cref="DataRecords"/>.
+        /// After aggregation the local buffer <c>currentInteractionData</c> is cleared.
+        /// </summary>
+        /// <param name="args">Gesture event arguments providing gesture state information.</param>
+        /// <exception cref="ArgumentException">Thrown if a grouped (joint, side) value does not map to a known segment.</exception>
+        protected void OnGestureCallback(HPUIGestureEventArgs args)
+        {
+            if (args.State == HPUIGestureState.Stopped)
+            {
+                foreach(IGrouping<(XRHandJointID, FingerSide), RaycastDataRecordsContainer> records in currentInteractionData.GroupBy(data => (data.handJointID, data.fingerSide)))
+                {
+                    HPUIInteractorConeRayAngleSegment segment = records.Key switch
```

Comment:
Please move this mapping outside the function, it does not pertain to the OnGestureCallback logic itself. Also consider making this mapping public, this is generally a pretty useful mapping to have for general HPUI interactions